autode/values.py

```diff
@@ -0,0 +1,203 @@
+
+    @abstractmethod
+    def __str__(self):
+        """String representation of this value"""
+
+    def copy(self):
+        return deepcopy(self)
+
+    def _convert_to_float(self, other):
+        """
+        Convert another value to these units, do nothing if not a Value
+
+        Arguments:
+            other (autode.values.Value | float):
+
+        Returns:
+            (float):
+        """
+
+        if not isinstance(other, Value):
+            return float(other)
+
+        return other.to(self.units).x
+
+    def __eq__(self, other):
+        """Equality of two values, which may be in different units
+        use default numpy close-ness to compare"""
+
+        if isinstance(other, Value):
+            if other.units == self.units:
+                return np.isclose(other.x, self.x)
+            else:
+                return np.isclose(other.to(self.units).x, self.x)
+
+        return np.isclose(other, self.x)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        """Less than comparison operator"""
+
+        if isinstance(other, Value):
+            return self.x < other.to(self.units).x
+
+        return self.x < other
+
+    def __gt__(self, other):
+        """Greater than comparison operator"""
+        return not self.__lt__(other)
+
+    def __le__(self, other):
+        """Greater than or equal to comparison operator"""
+        return self.__lt__(other) or self.__eq__(other)
+
+    def __ge__(self, other):
+        """Less than or equal to comparison operator"""
+        return self.__gt__(other) or self.__eq__(other)
+
+    def __add__(self, other):
+        """Add another value onto this one"""
+        return self.__class__(self.x + self._convert_to_float(other),
+                              units=self.units)
+
+    def __mul__(self, other):
+        """Multiply this value with another"""
+        return self.__class__(self.x * self._convert_to_float(other),
+                              units=self.units)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        return self.__add__(-other)
+
+    def __float__(self):
+        return self.x
+
+    def __int__(self):
+        return int(self.x)
+
+    def __complex__(self):
+        return complex(self.x)
+
+    def to(self,
+           units: Union[Unit, str]):
+        """Convert this value to a new unit, returning a copy
+
+        Arguments:
+            units (autode.units.Unit | str):
+
+        Returns:
+            (autode.values.Value):
+
+        Raises:
+            (TypeError):
+        """
+        if isinstance(units, str):
+
+            for imp_unit in self.implemented_units:
+                if units.lower() in imp_unit.aliases:
+                    units = imp_unit
+                    break
+
+        if units not in self.implemented_units:
```

Comment:
I think this goes through self.implemented_units twice. You could either do a "found_unit" bool, or do something like:
`try:
    imp_unit = one([unit for unit in self.implement_units if units.lower() in unit/aliases])
except (whatever the error is):
    raise TypeError(f'No viable type conversion from {self.units} '
                            f'-> {units}')`
I can't remember the exact syntax, but its something like that

Reply:
why has github killed my formatting :(((

Reply:
You've been away from Python for too long.. this should be fixed now

Reply:
I'm doing a python project currently! I just can't code without an ide ;)

Reply:
Yeah, what you did was what I mean :D
(except you used `next` instead of `one`, but I don't think that matters

Reply:
neither can I 🤣. Does `one` exist in Python?!

Reply:
its part of more_itertools